### PR TITLE
28 updates on add transaction view

### LINKED ISSUE #28: Updates on Add Transaction view
## Add Transaction Updates
- [x] Record Timestamp while capturing a transaction
- [x] Record Cash / Card while capturing a transaction
- [x] Use a more intuitive view (grids as suggested by @aashwath)
- [x] Open the form in modal
- [x] Autofocus the Amount field when adding a new transaction

### PR DESCRIPTION
**Pull request by usamazansari**
Closes #28

### Developer Checklist (Definition of Done)

*Issue*

- [x] All acceptance criteria from the issue are met
- [x] Tested in latest Chrome/Firefox

*Code*

- [x] Branch is up-to-date with the branch to be merged with #29 
- [x] Code is cleaned up and formatted

*PR*

- [x] Descriptive title for this pull request is provided (will be used for release notes later)
- [x] Reviewer and assignees are defined
- [x] Add type label (e.g., bug, feature) to this pull request
- [x] Add release label (e.g., `release: minor`) to this PR following [semver](https://semver.org/)
- [x] The PR is connected to the corresponding issue (via `Closes #...`)
- [x] [Summary of changes](#summary-of-changes) is written

### Summary of changes
- Add Transaction view as per the latest UI design

### Screenshots
- Desktop view
gnome-shell-screenshot-6P3CA2

- Mobile view
image

### Additional notes for the reviewer(s)
- This will be merged into update transaction list PR and then into develop

### CODE REVIEW

File: libs/core/src/lib/services/firestore/firestore-transaction/firestore-transaction.service.ts [hunk 1]
```diff
@@ -58,7 +65,7 @@ export class FirestoreTransactionService {
               id: docId,
               timestamp,
               amount: amount ?? 0,
```
Comment (usama251993) [inline libs/core/src/lib/services/firestore/firestore-transaction/firestore-transaction.service.ts:67]: Check if nullish coalescing can be removed

File: libs/shared/common/src/lib/components/switch/switch.component.ts [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+import { SelectWrapper } from '../select/select.types';
+import { FormControl } from '@angular/forms';
+
+@Component({
+  selector: 'expenses-tracker-switch',
+  standalone: true,
+  imports: [CommonModule],
+  templateUrl: './switch.component.html'
+})
+export class SwitchComponent<T = unknown> {
```
Comment (usama251993) [inline libs/shared/common/src/lib/components/switch/switch.component.ts:13]: Rename this to SegmentedControl

**Review by usama251993 — COMMENTED** [review]
Please take a look at the review comments

File: libs/core/src/lib/services/firestore/firestore-transaction/firestore-transaction.service.ts (hunk 1 above)
Comment (usamazansari) [reply libs/core/src/lib/services/firestore/firestore-transaction/firestore-transaction.service.ts]: Fixed in https://github.com/usamazansari/expenses-tracker/pull/74/commits/442489534fc8685d6df86656893e9bb0d8c81613

File: libs/shared/common/src/lib/components/switch/switch.component.ts (hunk 2 above)
Comment (usamazansari) [reply libs/shared/common/src/lib/components/switch/switch.component.ts]: Fixed in https://github.com/usamazansari/expenses-tracker/pull/74/commits/b2d2d85a0363ef4eef0444007fe9257f3e0cb5c3

**Review by usama251993 — APPROVED** [review]
Looks good!

**Pull request merged**